### PR TITLE
new program summary

### PR DESCRIPTION
**Pull request by tskille**
new program summary base on opm I/O classes. This program is similar to summary.x.

### CODE REVIEW

File: test_util/summary.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+static void printHelp() {
+
+    std::cout << "\ngetSmry needs minimum two argument. First is smspec filename and then list of vectors  \n"
+              << "\nIn addition, the program takes these options (which must be given before the arguments):\n\n"
+              << "-h Print help and exit.\n"
+              << "-l list all summary vectors.\n"
+              << "-r extrat data only for report steps. \n\n";
+}
+
+void printHeader(std::vector<std::string> keyList){
```
Comment (akva2) [inline test_util/summary.cpp]: pass by const ref.

**Comment by akva2** [comment]
jenkins run norne opm-tests=175 please

File: test_util/summary.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+#include <opm/io/eclipse/ESmry.hpp>
+
+
+static void printHelp() {
+
+    std::cout << "\ngetSmry needs minimum two argument. First is smspec filename and then list of vectors  \n"
+              << "\nIn addition, the program takes these options (which must be given before the arguments):\n\n"
+              << "-h Print help and exit.\n"
+              << "-l list all summary vectors.\n"
+              << "-r extrat data only for report steps. \n\n";
```
Comment (akva2) [inline test_util/summary.cpp]: extract

File: test_util/summary.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+#include <tuple>
+#include <getopt.h>
+#include <sstream>
+  
+#include <opm/io/eclipse/ESmry.hpp>
+
+
+static void printHelp() {
+
+    std::cout << "\ngetSmry needs minimum two argument. First is smspec filename and then list of vectors  \n"
```
Comment (akva2) [inline test_util/summary.cpp]: getSmry is in fact a better, less generic name for the binary rather than 'summary'

File: test_util/summary.cpp [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        return 0;
+    }
+
+    std::vector<std::string> smryList;
+    for (int i=0; i<argc - argOffset-1;i++){
+       smryList.push_back(argv[i+argOffset+1]);
+    }
+    
+    if (smryList.size()==0){
+        std::string message = "No summary keys spesified on command line";
```
Comment (akva2) [inline test_util/summary.cpp]: specified

File: src/opm/io/eclipse/ESmry.cpp [hunk 5]
```diff
@@ -31,6 +31,7 @@
 #include <set>
 #include <stdexcept>
 
+#include <iostream>
```
Comment (akva2) [inline src/opm/io/eclipse/ESmry.cpp]: drop this file from this PR

File: test_util/summary.cpp (hunk 3 above)
Comment (akva2) [reply test_util/summary.cpp]: on alternaive is maybe summaryEcl to be in line with the others ?

Comment (alfbr) [reply test_util/summary.cpp]: summary is available, why not take it? Compare was taken by imagemagick, that's why the awkward ECL is there.

Comment (akva2) [reply test_util/summary.cpp]: did not know it was a contest ;) i thought it was about making it clear to the user what a binary does.

Comment (alfbr) [reply test_util/summary.cpp]: Users think of the files as summary files. What is clear from a programmer is often not clear for the user.  For instance, the application for commandline plotting is summaryplot, not summaryEclPlot. What is a good variable naming convention for a programmer is often not a good naming convention for user applications.

Comment (akva2) [reply test_util/summary.cpp]: but but but then users are wrong! ;p i really don't care, keep it as summary then.

Comment (alfbr) [reply test_util/summary.cpp]: Cool, thanks! This one is not a biggie for me either, but I do think it was super good that we chose the generic name Flow a few years back.

**Comment by akva2** [comment]
argh, my bad. u have to add the new file :/

**Comment by bska** [comment]
jenkins run norne opm-tests=175 please

**Comment by tskille** [comment]
I have now re-based the branch tskille:summary.  Notice that this PR also depends on PR #175 in opm tests   (plotwells.sh in norne now using summary).

Please let me know if you need more from my end.

**Comment by akva2** [comment]
it's my fault but the new file is not in git. you have to git add it.

**Comment by tskille** [comment]
sorry, I did not notice. It should be there now. Thanks

**Comment by akva2** [comment]
jenkins run norne opm-tests=175 please

**Comment by akva2** [comment]
tested working from downstream. merging. thanks!

**Pull request merged**